qunicorn_core/core/job_service.py

```diff
@@ -101,10 +104,20 @@ def send_job_to_pilot():
     raise NotImplementedError
 
 
-def cancel_job_by_id(job_id):
+def cancel_job_by_id(job_id, token):
     """cancel job execution"""
-    # TODO: Implement Cancel
-    raise NotImplementedError
+    job: JobDataclass = job_db_service.get_job_by_id(job_id)
+    job_request: JobRequestDto = job_mapper.dataclass_to_request(job)
+    job_core_dto: JobCoreDto = job_mapper.request_to_core(job_request)
```

Comment:
Hier wäre ein mapper sinnvoll, der dataclass in core mapped (wenns den noch nicht gibt) 
Dann brauchst du auch Zeile 11 - 116 nicht.